packages/client-proxy/main.go

```diff
@@ -52,103 +64,196 @@ func run() int {
 			zap.L().Fatal("failed to create metrics exporter", zap.Error(err))
 		}
 	}
+
 	defer func() {
 		err := tel.Shutdown(ctx)
 		if err != nil {
 			log.Printf("telemetry shutdown:%v\n", err)
 		}
 	}()
 
-	logger := zap.Must(e2bLogger.NewLogger(ctx, e2bLogger.LoggerConfig{
-		ServiceName: serviceName,
-		IsInternal:  true,
-		IsDebug:     env.IsDebug(),
-		Cores:       []zapcore.Core{e2bLogger.GetOTELCore(tel.LogsProvider, serviceName)},
-	}))
+	logger := zap.Must(
+		e2bLogger.NewLogger(
+			ctx, e2bLogger.LoggerConfig{
+				ServiceName: serviceName,
+				IsInternal:  true,
+				IsDebug:     env.IsDebug(),
+				Cores:       []zapcore.Core{e2bLogger.GetOTELCore(tel.LogsProvider, serviceName)},
+			},
+		),
+	)
+
 	defer func() {
 		err := logger.Sync()
 		if err != nil {
 			log.Printf("logger sync error: %v\n", err)
 		}
 	}()
+
 	zap.ReplaceGlobals(logger)
 
+	proxyPort := internal.GetProxyServicePort()
+	edgePort := internal.GetEdgeServicePort()
+	edgeSecret := internal.GetEdgeServiceSecret()
+	orchestratorPort := internal.GetOrchestratorServicePort()
+
+	exitCode := atomic.Int32{}
+
+	wg := sync.WaitGroup{}
+
+	signalCtx, sigCancel := signal.NotifyContext(ctx, syscall.SIGTERM, syscall.SIGINT)
+	defer sigCancel()
+
 	logger.Info("Starting client proxy", zap.String("commit", commitSHA), zap.String("instance_id", instanceID))
 
-	healthServer := &http.Server{Addr: fmt.Sprintf(":%d", healthCheckPort)}
-	healthServer.Handler = http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
-		writer.WriteHeader(http.StatusOK)
-	})
+	tracer := tel.TracerProvider.Tracer(serviceName)
 
-	wg.Add(1)
-	go func() {
-		// Health check
-		defer wg.Done()
+	edgeSD, orchestratorsSD, err := service_discovery.NewServiceDiscoveryProvider(ctx, edgePort, orchestratorPort, logger)
+	if err != nil {
+		logger.Error("failed to resolve service discovery config", zap.Error(err))
+		return 1
+	}
 
-		logger.Info("starting health check server", zap.Int("port", healthCheckPort))
-		err := healthServer.ListenAndServe()
-		switch {
-		case errors.Is(err, http.ErrServerClosed):
-			logger.Info("http service shutdown successfully", zap.Int("port", healthCheckPort))
-		case err != nil:
-			exitCode.Add(1)
-			logger.Error("http service encountered error", zap.Int("port", healthCheckPort), zap.Error(err))
-		default:
-			// this probably shouldn't happen...
-			logger.Error("http service exited without error", zap.Int("port", healthCheckPort))
+	var redisClient *redis.Client
+	var catalog sandboxes.SandboxesCatalog
+
+	redisUrl := os.Getenv("REDIS_URL")
```

Comment:
can you support redis cluster as here:
https://github.com/e2b-dev/infra/blob/main/packages/api/internal/handlers/store.go#L119